azure-pipelines.yml

```diff
@@ -0,0 +1,69 @@
+# nimi-bot pipeline for running all system tests as part of Continuous Integration.
+
+parameters:
+- name: module_names
+  type: object
+  default: # Order of the modules is based on test execution time, longest firt.
+  - nidigital
+  - nitclk  
+  - nifgen
+  - nidcpower
+  - nidmm
+  - niscope
+  - nimodinst
+  - nise
+  - niswitch
+
+# A pipeline with no CI trigger
+trigger: none
+
+# specific path
+pr:
+  branches:
+    include:
+    - master
+    - releases/*
+  paths:
+    exclude:
+    - CHANGELOG.md
+    - CONTRIBUTING.md
+    - .gitattributes
+    - LICENSE
+    - VERSION
+  drafts: false
+
+stages:
+
+  - stage: ensure_codegen_up_to_date
```

Comment:
@sbethur - this is a good check to have in place, but seems strange to me to run this on nimi-bot as it is not a system test.
We run unit tests, linter tests, codecoverage on TravisCI and this seems closer to those. No? What is our long term plan for all that?

@SukruthGrandhiNI - this is a good check to have in place, but it seems orthogonal to this PR. Can we make this addition in its own PR?

Reply:
@marcoskirsch I think we can have it here because we are running system tests on the generated folders. so we can validate generated folders before running the system tests.
I am also fine removing this check from this PR.

Reply:
The long term plan is to do all the steps in a single pipeline using GitHub Actions. Till we get there I'm not too concerned about where we're doing this step, but having it somewhere is definitely a net positive. If adding to TravisCI is not trivial, then it might not be worth the time since we want to move away from it.

Either way, it would be ideal to do it in its own PR.

Reply:
I have removed this check from this PR